apps/prairielearn/src/lib/grading.js

```diff
@@ -14,137 +11,98 @@ import { getQuestionCourse } from './question-variant';
 import * as sqldb from '@prairielearn/postgres';
 import * as questionServers from '../question-servers';
 import * as workspaceHelper from './workspace';
+import { IdSchema } from './db-types';
 
 const sql = sqldb.loadSqlEquiv(__filename);
 
-/**
- * Internal error type for tracking lack of submission.
- */
-class NoSubmissionError extends Error {
-  constructor(message) {
-    super(message);
-    this.name = 'NoSubmissionError';
-  }
-}
```

Comment:
This was a workaround required with `async.series`, but that can be easily be replaced with a return statement instead now, so this class is no longer needed.